src/System.Linq/tests/SelectTests.cs

```diff
@@ -0,0 +1,230 @@
+﻿
+using System.Collections.Generic;
+using Xunit;
+
+namespace System.Linq.Tests
+{
+    public class SelectTests
+    {
+        [Fact]
+        public void Select_SourceIsNull_ArgumentNullExceptionThrown()
+        {
+            IEnumerable<int> source = null;
+            Func<int, int> selector = i => i + 1;
+
+            AssertException<ArgumentNullException>(() => source.Select(i => i + 1));
```

Comment:
Please feel free to add System.Threading.Tasks to the packages.config for System.Linq.Tests.  We're fairly strict about what gets added to packages.config for the product assemblies, but we're much more liberal with the tests.